tests/epyccel/test_epyccel_division.py

```diff
@@ -137,50 +137,80 @@ def test_call_fdiv_i_i(language):
     def fdiv_i_i(x, y):
         return x // y
 
-    f = epyccel(fdiv_i_i, language=language)
+    fflags = "-Werror -Wconversion"
+
+    f = epyccel(fdiv_i_i, language=language, fflags=fflags)
     x = randint(1e9)
     y = randint(low=1, high= 1e3)
 
     assert (f(x, y) == fdiv_i_i(x, y))
     assert (f(-x, y) == fdiv_i_i(-x, y))
     assert (f(x, -y) == fdiv_i_i(x, -y))
     assert (f(-x, -y) == fdiv_i_i(-x, -y))
+    assert isinstance(f(x, y), type(fdiv_i_i(x, y)))
+
+def test_call_fdiv_i_i_i(language):
+    @types(int, int, int)
+    def fdiv_i_i_i(x, y, z):
+        return x // y // z
+
+    fflags = "-Werror -Wconversion"
+
+    f = epyccel(fdiv_i_i_i, language=language, fflags=fflags)
+    x = randint(1e9)
+    y = randint(low=1, high= 1e3)
+    z = randint(low=1, high= 1e2)
+
+    assert (f(x, y, z) == fdiv_i_i_i(x, y, z))
+    assert (f(-x, y, z) == fdiv_i_i_i(-x, y, z))
+    assert (f(x, -y, z) == fdiv_i_i_i(x, -y, z))
+    assert (f(-x, -y, z) == fdiv_i_i_i(-x, -y, z))
+    assert isinstance(f(x, y, z), type(fdiv_i_i_i(x, y, z)))
 
 def test_call_fdiv_i_r(language):
     @types(int, 'real')
     def fdiv_i_r(x, y):
         return x // y
 
-    f = epyccel(fdiv_i_r, language=language)
+    fflags = "-Werror -Wconversion"
+
+    f = epyccel(fdiv_i_r, language=language, fflags=fflags)
     x = randint(1e9)
     y = uniform(low=1, high= 1e3)
     assert (f(x, y) == fdiv_i_r(x, y))
     assert (f(-x, y) == fdiv_i_r(-x, y))
     assert (f(x, -y) == fdiv_i_r(x, -y))
     assert (f(-x, -y) == fdiv_i_r(-x, -y))
+    assert isinstance(f(x, y), type(fdiv_i_r(x, y)))
 
 def test_call_fdiv_r_i(language):
     @types('real', int)
     def fdiv_r_i(x, y):
         return x // y
 
-    f = epyccel(fdiv_r_i, language=language)
+    fflags = "-Werror -Wconversion"
+
+    f = epyccel(fdiv_r_i, language=language, fflags=fflags)
     x = uniform(high=1e9)
     y = randint(low=1, high= 1e3)
     assert (f(x, y) == fdiv_r_i(x, y))
     assert (f(-x, y) == fdiv_r_i(-x, y))
     assert (f(x, -y) == fdiv_r_i(x, -y))
     assert (f(-x, -y) == fdiv_r_i(-x, -y))
+    assert isinstance(f(x, y), type(fdiv_r_i(x, y)))
 
 def test_call_fdiv_r_r(language):
     @types('real', 'real')
     def fdiv_r_r(x, y):
         return x // y
 
-    f = epyccel(fdiv_r_r, language=language)
+    fflags = "-Werror -Wconversion"
+
+    f = epyccel(fdiv_r_r, language=language, fflags=fflags)
     x = uniform(high=1e9)
     y = uniform(low=1e-14, high= 1e3)
     assert (f(x, y) == fdiv_r_r(x, y))
     assert (f(-x, y) == fdiv_r_r(-x, y))
     assert (f(x, -y) == fdiv_r_r(x, -y))
     assert (f(-x, -y) == fdiv_r_r(-x, -y))
+    assert isinstance(f(x, y), type(fdiv_r_r(x, y)))
```

Comment:
Given my comment above I want to ask for 1 or 2 tests with numpy types.

Reply:
mm, I don't know what you mean exactly.
do you mean using numpy types in the random values, or using precise types like (int8, int16 ...) in the types decorate?.

Reply:
Both. Using precise types in the types decorator, which then means you need to pass random values with the matching type

Reply:
ok, the same as what I did with numpy array as function arguments test.